api.go

```diff
@@ -1002,6 +1007,114 @@ func marshalResponse(resp interface{}, w http.ResponseWriter, status int, r *htt
 	return nil
 }
 
+func filterSparseFields(resp interface{}, r *http.Request) (interface{}, error) {
+	query := r.URL.Query()
+	queryParams := parseQueryFields(&query)
```

Comment:
so we actually do a regexp check now on any request? can we check beforehand if its actually necessary? Might not perform that good, what do you think?